packages/core/lib/src/core_data.dart

```diff
@@ -12,18 +12,18 @@ part 'data/text_style.dart';
 /// A building element metadata.
 abstract class BuildMetadata {
   /// The associatd element.
-  final dom.Element element;
+  final dom.Element? element;
 
   final TextStyleBuilder _tsb;
 
   /// Creates a node.
   BuildMetadata(this.element, this._tsb);
 
   /// The registered build ops.
-  Iterable<BuildOp> get buildOps;
+  Iterable<BuildOp>? get buildOps;
```

Comment:
I think this should be non-nullable.

Reply:
I tried, and almost all tests fail if you do that. If you look at [BuildTree._collectMetadata()](https://github.com/daohoangson/flutter_widget_from_html/pull/436/files#diff-5eefa0a05ac4e36c101e4abd26251c8ff32f6edf8bb9afca3c8236a285c88ff7R209) it does null checks and as the tests show, `buildOps` and `parentOps` can in fact be null in this case.

We could definitely use an empty list as the default value instead of null, if you prefer.

Reply:
Changed it to use empty lists as a default value.